src/pagination/pagination.tsx

```diff
@@ -0,0 +1,214 @@
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const paginationItemList = [...Array(totalPages)].map((e, i) => {
+        const pageIndex = i + 1;
+        const boundaryRangeLocal =
+            boundaryRange && boundaryRange > 0 ? boundaryRange : 1;
+        const siblingRangeLocal =
+            siblingRange && siblingRange >= 0 ? siblingRange : 0;
+        const startRangeLocal = boundaryRangeLocal + 2 + siblingRangeLocal * 2;
+        const endRange = totalPages - startRangeLocal;
+        const totalRange = (boundaryRangeLocal + siblingRangeLocal) * 2 + 3;
+        const active = activePage === pageIndex;
+        if (totalPages <= totalRange) {
+            return (
+                <Item
+                    key={pageIndex}
+                    onClick={() => handlePaginationItemOnClick(pageIndex)}
+                    $selected={active}
+                >
+                    {pageIndex}
+                </Item>
+            );
+        }
+
+        const ellipsisStart =
+            activePage + siblingRangeLocal > startRangeLocal &&
+            pageIndex === boundaryRangeLocal + 1;
+        const ellipsisEnd =
+            activePage - siblingRangeLocal <= endRange &&
+            pageIndex === totalPages - boundaryRangeLocal - 1;
+
+        if (ellipsisStart || ellipsisEnd) {
+            return (
+                <EllipsisItem key={pageIndex}>
+                    <EllipsisHorizontal $disabled={false} />
+                </EllipsisItem>
+            );
+        }
+
+        const paginationStart =
+            (pageIndex <= startRangeLocal &&
+                activePage + siblingRangeLocal <= startRangeLocal) ||
+            pageIndex <= boundaryRangeLocal;
+        const paginationMiddle =
+            pageIndex === activePage ||
+            (pageIndex <= activePage + siblingRangeLocal &&
+                pageIndex >= activePage - siblingRangeLocal);
+        const paginationEnd =
+            (pageIndex > endRange &&
+                activePage - siblingRangeLocal > endRange) ||
+            pageIndex > totalPages - boundaryRangeLocal;
+        if (paginationStart || paginationMiddle || paginationEnd) {
+            return (
+                <Item
+                    key={pageIndex}
+                    onClick={() => handlePaginationItemOnClick(pageIndex)}
+                    $selected={active}
+                >
+                    {pageIndex}
+                </Item>
+            );
+        }
+
+        return null;
+    });
+
+    return (
+        <PaginationWrapper
+            className={className}
+            ref={ref}
+            id={id || "pagination-wrapper"}
+            data-testid={dataTestId || "pagination"}
+        >
+            <PaginationList>
+                <PaginationMenu>
+                    {showFirstAndLastNav && (
+                        <LinkIcon
```

Comment:
for the navigation items, can add a `aria-label` and `tabIndex` of -1 (disabled) or 0 (enabled) so that they are focusable with the tab key